### PR TITLE
refactor: updated mechanism which allows maps to track each other.

### PR DESCRIPTION
**Pull request by DanielLyleWilliamWarren**
- Using timeout to create a better user experience when tracking maps.

### CODE REVIEW

File: src/app/pages/quickMaps/pages/dietaryChange/components/comparisonCard/scenariosMap/scenariosMap.component.ts [hunk 1]
```diff
@@ -98,45 +94,23 @@ export class ScenariosMapComponent implements AfterViewInit {
...
-    if (targetMap.getCenter() !== baseMap.getCenter()) {
-      targetMap.panTo(baseMap.getCenter());
-    }
-  }
-
-  private setMapZoom(baseMap: L.Map, targetMap: L.Map): void {
-    if (targetMap.getZoom() !== baseMap.getZoom()) {
-      targetMap.setZoom(baseMap.getZoom());
-    }
+  private alignMaps(baseMap: L.Map, targetMap: L.Map): void {
```
Comment (jon571) [inline src/app/pages/quickMaps/pages/dietaryChange/components/comparisonCard/scenariosMap/scenariosMap.component.ts:108]: You might want to change the name of the variable here as "baseMap" is a bit too similar to "baseline". 
I dunno what to :-)

primary and secondary?
control and mirror?
changing and repeating?

File: src/app/pages/quickMaps/pages/dietaryChange/components/comparisonCard/scenariosMap/scenariosMap.component.ts [hunk 2]
```diff
@@ -98,45 +94,23 @@ export class ScenariosMapComponent implements AfterViewInit {
 
   private initialiseListeners(): void {
```
Comment (jon571) [inline src/app/pages/quickMaps/pages/dietaryChange/components/comparisonCard/scenariosMap/scenariosMap.component.ts:95]: I think you should still have your move and zoom listeners as they were, making it reflect changes straight away.  I just think that after they have called `setMapZoom` or `setMapPosition` they should then call your `alignMaps` function to ensure that everything is aligned after all zooms and movements are done.

Comment (DanielLyleWilliamWarren) [reply src/app/pages/quickMaps/pages/dietaryChange/components/comparisonCard/scenariosMap/scenariosMap.component.ts:95]: I had a zoom listener before, something to do with a trigger on animation was causing it so jump around a bit if it wasn't used, that's why the mousedown() listener was being used too. But as I am now using setView() with animation disabled, the move() listener is called when user zooms too, and it is working properly.

Comment (jon571) [reply src/app/pages/quickMaps/pages/dietaryChange/components/comparisonCard/scenariosMap/scenariosMap.component.ts:95]: So are you happy with the mirrored map not aligning until the user stops moving then? I thought that was quite nice.

Comment (DanielLyleWilliamWarren) [reply src/app/pages/quickMaps/pages/dietaryChange/components/comparisonCard/scenariosMap/scenariosMap.component.ts:95]: It still mirrors and is more responsive than the current implementation as zoom all pan animations are disabled. You can test the current implementation [here](https://develop--micronutrientsupport-tool.netlify.app/quick-maps/diet/dietary-change?country-id=NGA&mnd-id=C&measure=diet&data-level=country&age-gender-group=all)

Comment (jon571) [reply src/app/pages/quickMaps/pages/dietaryChange/components/comparisonCard/scenariosMap/scenariosMap.component.ts:95]: I've added @bgsandan as a reviewer to give the stamp of approval.

Comment (bgsandan) [reply src/app/pages/quickMaps/pages/dietaryChange/components/comparisonCard/scenariosMap/scenariosMap.component.ts:95]: Looks good to me :+1:

**Review by jon571 — APPROVED** [review]

**Pull request merged**